### PR TITLE
fix: track and display lastAction on player so freeze and slash show instead of claim

### PR DESCRIPTION
**Pull request by ldeffenb**

### CODE REVIEW

File: src/types/entities/player.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export class Player {
 	private line: number // where this player is in the players list TODO: -1 for not visible
 	private _isPlaying = false
 	private lastBlock: BlockDetails | undefined // block details of last interaction
+	private lastAction: string | undefined // tracks last action which cannot be derived from blockNumber
```
Comment (mfw78) [inline src/types/entities/player.ts:25]: I'd most probably suggest that given that the actions are limited and defined, that an enum should be used for this as opposed to a string to more strictly enforce typing.

File: src/types/entities/player.ts [hunk 2]
```diff
@@ -138,6 +138,7 @@ export class Player {
 	 */
 	commit(block: BlockDetails) {
 		this.lastBlock = block
+		this.lastAction = 'commit'
```
Comment (mfw78) [inline src/types/entities/player.ts:141]: Use enum here

File: src/types/entities/player.ts [hunk 3]
```diff
@@ -121,7 +120,8 @@ export class Player {
 		let t = `${Round.roundString(
 			this.lastBlock!.blockNo
 		)} ${this.overlayString()}`
-		t += ` ${Round.roundPhaseFromBlock(this.lastBlock!.blockNo)}`
+		if (this.lastAction) t += ` ${this.lastAction}`
```
Comment (mfw78) [inline src/types/entities/player.ts:123]: Use enum here

File: src/types/entities/player.ts [hunk 4]
```diff
@@ -151,6 +152,7 @@ export class Player {
 
 	reveal(block: BlockDetails, round: number, hash: string, depth: number) {
 		this.lastBlock = block
+		this.lastAction = 'reveal'
```
Comment (mfw78) [inline src/types/entities/player.ts:155]: Use enum here

File: src/types/entities/player.ts [hunk 5]
```diff
@@ -162,6 +164,7 @@ export class Player {
 	 */
 	claim(block: BlockDetails, _amount: BigNumber) {
 		this.lastBlock = block
+		this.lastAction = 'claim'
```
Comment (mfw78) [inline src/types/entities/player.ts:167]: Use enum here

File: src/types/entities/player.ts [hunk 6]
```diff
@@ -190,14 +194,15 @@ export class Player {
 
 	updateStake(block: BlockDetails, amount: BigNumber) {
 		this.lastBlock = block
+		this.lastAction = 'stake'
```
Comment (mfw78) [inline src/types/entities/player.ts:197]: Use enum here

File: src/types/entities/player.ts [hunk 7]
```diff
@@ -176,11 +179,12 @@ export class Player {
 	 */
 	freeze(block: BlockDetails, thawBlock: number) {
 		this.lastBlock = block
+		this.lastAction = 'freeze'
```
Comment (mfw78) [inline src/types/entities/player.ts:182]: Use enum here

File: src/types/entities/player.ts [hunk 8]
```diff
@@ -213,6 +218,7 @@ export class Player {
 	 */
 	slash(block: BlockDetails, amount: BigNumber) {
 		this.lastBlock = block
+		this.lastAction = 'slash'
```
Comment (mfw78) [inline src/types/entities/player.ts:221]: Use enum here

**Review by mfw78 — CHANGES_REQUESTED** [review]
Overall looks good, just should use enum for stricter type enforcement.

**Comment by ldeffenb** [comment]
While I agree an enum for the phase would be good, can we push this one into develop so that I can get more PRs put together for my other hacks updates?   I'd like to have a good, working, clean, enhanced version ready before RC8 hits mainnet which I'm suspecting will be right around the call next week.

**Comment by mfw78** [comment]
No problems pushing it through, just open an issue to do the enum so we don't forget 😊

**Pull request merged**